sound/soc/sof/intel/mtl.c

```diff
@@ -555,9 +556,9 @@ static irqreturn_t mtl_ipc_irq_thread(int irq, void *context)
 			sdev->ipc->msg.rx_data = &notification_data;
 			snd_sof_ipc_msgs_rx(sdev);
 			sdev->ipc->msg.rx_data = NULL;
-		}
 
-		mtl_ipc_host_done(sdev);
+			mtl_ipc_host_done(sdev);
+		}
```

Comment:
I wasn't able to figure out what this change fixes. The commit message is a bit too cryptic to me.

Reply:
Moving the reply acking fixes this by making sure that we only send a new message after we have certainly acked the reception of the reply.

With the fix on the firmware side, this should not be needed anymore, but I rather play safe.